tamr_unify_client/models/attribute/collection.py

```diff
@@ -0,0 +1,96 @@
+from tamr_unify_client.models.attribute.resource import Attribute
+from tamr_unify_client.models.base_collection import BaseCollection
+
+
+class AttributeCollection(BaseCollection):
+    """Collection of :class:`~tamr_unify_client.models.attribute.resource.Attribute` s.
+
+    :param client: Client for API call delegation.
+    :type client: :class:`~tamr_unify_client.Client`
+    :param :py:class:`dict` data: JSON data representing this resource
+    :param api_path: API path used to access this collection.
+        E.g. ``"datasets/1/attributes"``.
+    :type api_path: str
+    """
+
+    def __init__(self, client, data, api_path):
+        super().__init__(client, api_path)
+        self._data = data
+
+    @classmethod
+    def from_json(cls, client, data, api_path):
+        # BaseCollection doesn't really implement from_json / from_data
+        # but we pretend it does.
+        return AttributeCollection(client, data, api_path)
+
+    def by_resource_id(self, resource_id):
+        """Retrieve an attribute by resource ID.
+
+        :param resource_id: The resource ID. E.g. ``"AttributeName"``
+        :type resource_id: str
+        :returns: The specified attribute.
+        :rtype: :class:`~tamr_unify_client.models.attribute.resource.Attribute`
+        """
+        return self.by_name(resource_id)
+
+    def by_relative_id(self, relative_id):
+        """Retrieve an attribute by relative ID.
+
+        :param relative_id: The resource ID. E.g. ``"datasets/1/attributes/AttributeName"``
+        :type relative_id: str
+        :returns: The specified attribute.
+        :rtype: :class:`~tamr_unify_client.models.attribute.resource.Attribute`
+        """
+        split_id = relative_id.split("/")
+        if split_id[:-1] != self.api_path:
```

Comment:
This might be too aggressive.
For example, `datasets/1/attributes` might be the same as `projects/1/unifiedDataset/attributes`.